### DISCUSSION TITLE
Long running processes should not be run in webinterface

### DISCUSSION THREAD

**Discussion by amtriorix** [PrestaShop/PrestaShop#34677] [Ideas]
All those processes:
1) upgrade of prestashop, including 1 click upgrade
2) import data
3) export data
4) sitemap generation
5) seo keyword generation
6) creating/clone store from another store (with data copy)

Should NOT run in full using the webinterface. It should be programmed it runs with symphony console commands and using php-cli that signals the state of the process, (idle, run, error, pause, progress)  to a table in the database.

The webinterface should only be used to interact with that table in the database, however the processing should run inside php-cli.

Doing so timeouts will be avoided and there is no need to run to hope, wait and have crossed fingers so a long process terminates in good order. But mostly not when the process exceeds a lot of time (very often php script timeout) or excess data.

Again. A php script should not run for more then a limited time for the stability of an environment. Rule of thumb, when a script runs for more then 600 sec, it should be running in a dispatcher, queued with event signaling. 

Using ajax calls and let a script run for too long in the backend is bad practice.

However not all work in this regard is useless, but the webgui should reflect the state of the process, shows event logs and finally should interact with the process to start, end, pause, resume etc as a control board so to speak to the process that runs in php-cli.

Right now modules like 1 click upgrade do mess up a big site with a lot of data.

If you want to ruin, to break a big prestashop store with multistores, well use 1 click upgrade and you are in deep trouble the way it is designed.

**Also I see no reason why prestashop core and module updates should not be done using composer update as this is the way it should be done that a lot of prestashop code should be in packages.**

Why to make it difficult as there can be a (git based) vendor/prestashop/core package, including vendor/prestashop/modules/<coder>/<module-name> package.

Packages are designed to make upgrades for a big part more easy
ref: https://symfony.com/blog/upgrading-your-symfony-projects-the-easy-way

Unfortionally prestashop is not written in laravel or even data migrations of sql tables would be a snap using: https://laravel.com/docs/5.7/artisan

Additionnal information
PrestaShop version: 1.7.4
PHP version: 7.2

**Comment by khouloudbelguith** [comment]
Hi @amtriorix,

Thanks for your report!
@eternoendless @PrestaShop/prestashop-core-developers what do you think about his suggestion?

Thanks!

**Comment by hibatallahAouadni** [comment]
Hello,

This improvement has been open since 2018 and no one is reacting to it.
Ping @PrestaShop/product-team could you please take a look. Should I close it? what do you think?

Thanks!

**Comment by matks** [comment]
Hi @hibatallahAouadni this improvement is a very nice idea but it will take years to implement it. It'll be done step by step, little by little. It would be a major architectural change so this is not something that can be done easily. But it's definitely a good thing, a good idea.

**Comment by LouiseBonnard** [comment]
@matks, what do you suggest regarding this ticket then: should add to the next major kanban or divide this ticket into smaller tasks we will be able to prioritize version by version and close it?

**Comment by matks** [comment]
Hi, this task is for far in the future I think we can keep it single for now, and split it when we can come to it.

> Also I see no reason why prestashop core and module updates should not be done using composer update as this is the way it should be done that a lot of prestashop code should be in packages.

https://github.com/PrestaShop/PrestaShop/issues/12586
